src/mlpack/methods/reinforcement_learning/replay/episodic_replay.hpp

```diff
@@ -0,0 +1,214 @@
+ * @tparam EnvironmentType Desired task.
+ */
+template <typename EnvironmentType>
+class EpisodicReplay
+{
+ public:
+  //! Convenient typedef for action.
+  using ActionType = typename EnvironmentType::Action;
+
+  //! Convenient typedef for state.
+  using StateType = typename EnvironmentType::State;
+
+  EpisodicReplay():
+      capacity(0),
+      position(0),
+      full(false)
+  { /* Nothing to do here. */ }
+
+  /**
+  * Construct an instance of episode experience replay class.
+  *
+  * @param capacity Total memory size in terms of number of episodes.
+  * @param dimension The dimension of an encoded state.
+  */
+
+  EpisodicReplay(const size_t capacity,
+                 const size_t dimension = StateType::dimension) :
+      capacity(capacity),
+      states.resize(capacity),
+      next_states.resize(capacity),
+      rewards.resize(capacity),
+      actions.resize(capacity),
+      isTerminal.resize(capacity),
+      position(0),
+      full(false)
+  { /* Nothing to do here. */ }
+
+
+  /**
+  * Store the given experience.
+  *
+  * @param state Given state.
+  * @param action Given action.
+  * @param reward Given reward.
+  * @param nextState Given next state.
+  * @param isEnd Whether next state is terminal state.
+  */
+  void Store(const StateType& state,
+	     ActionType action,
+	     double reward,
+	     const StateType& nextState,
+	     bool isEnd)
+  {
+    states[position].push_back(state.Encode());
+    actions[position].push_back(action);
+    rewards[position].push_back(reward);
+    nextStates[position].push_back(nextState.Encode());
+    isTerminal[position].push_back(isEnd);
+    if(isTerminal){
+      position++;
+    }
+    if (position == capacity)
+    {
+      full = true;
+      position = 0;
+    }
+  }
+
+  /**
+  * Sample some episodes.
+  *
+  * @param sampledStates Sampled encoded states.
+  * @param sampledActions Sampled actions.
+  * @param sampledRewards Sampled rewards.
+  * @param sampledNextStates Sampled encoded next states.
+  * @param isTerminal Indicate whether corresponding next state is terminal
+  *        state.
+  */
+
+  void Random_Episode(arma::mat& episodeStates,
+                      arma::icolvec& episodeActions,
+	              arma::colvec& episodeRewards,
+	              arma::mat& episodeNextStates,
+	              arma::icolvec& isTerminal)
+  {
+    size_t upperBound = full ? capacity : position;
+    srand(time(NULL));
+    int episodeNum = rand()%(upperBound);
+    vector<arma::colvec> temp = states[episodeNum];
+    int i=0;
+    for(auto state : temp){
+      if(i==0){
+        episodeStates = state;
+        i++;
+      }else{
+        episodeStates = arma::join_rows(episodeStates,state);
+      }
+    }
+    episodeActions = arma::conv_to<arma::icolvec>::from(actions[episodeNum]);
+    episodeRewards = arma::conv_to<arma::icolvec>::from(rewards[episodeNum]);
+    temp = next_states[episodeNum];
+    int i=0;
+    for(auto state : temp){
```

Comment:
This can be written as:
> for (auto state : temp)
>  {
      if (i == 0)
>    {
>      episodeNextStates = state;
>      i++;
>    }
> }
but the if's must be 2 spaces tab for each indentation.
You can refer to the guideline as given in above comment.
Thanks.
Thanks.